### DISCUSSION TITLE
TAC avec le nouveau SDK Piano Analytics

### DISCUSSION THREAD

**Discussion by JLBWai** [AmauriC/tarteaucitron.js#1114] [General]
Bonjour Amauri, 

J'ai une question concernant le paramétrage de TAC avec Piano Analytics. 
Est-ce qu'il est possible de gérer le consentement via TAC pour le nouveau SDK Piano (PA Tagging) ? 
Je peux voir le service AT Internet disponible mais pour le **marquage Smarttag** , qui est désormais "**obsolète**" dans la roadmap Piano.

Merci.

**Comment by AmauriC** [comment]
Bonjour : )

Quel est le marqueur HTML du SDK Piano?

**Reply by JLBWai** [reply]
Voici le marqueur, avec exemple d'événement de page.display

<html>
  <head>
    <title>My Page</title>
    <script src="https://tag.aticdn.net/piano-analytics.js"></script>
    <script type="text/javascript">
      pa.setConfigurations({ // Basic configuration to send events
        site:123456789,
        collectDomain:'https://logsx.xiti.com'
      }); 
    </script>
  </head>
  <body>
      ...
    <script type="text/javascript">
      pa.sendEvent('page.display', { // Event name
        'page' : 'page name',
        'page_chapter1' : 'level 1',
        'page_chapter2' : 'level 2',
        'page_chapter3' : 'level 3'
      });
    </script>
  </body>
</html>

**Reply by AmauriC** [reply]
Merci, j'ai ouvert une issue https://github.com/AmauriC/tarteaucitron.js/issues/1115, j'essai de m'en occuper rapidement.

**Reply by JLBWai** [reply]
Bonjour Amauri, 

Je me permets de revenir vers vous. Avez-vous pu avancer sur la configuration du nouveau SDK Piano pour TAC ? 

Merci

**Reply by AmauriC** [reply]
J'essai de m'en occuper au w-e ; )

**Reply by JLBWai** [reply]
Bonjour Amauri, 

Je suis allée voir sur TAC, et je peux voir que le service Piano Analytics est désormais disponible. Cela signifie-t-il qu'on peut l'utiliser pour gérer le consentement via les nouveaux SDK Piano ? 

Merci d'avance.

**Reply by AmauriC** [reply]
Oui le service est bien dispo 👌
Je n'ai pas eu de retour sur l'utilisation, si vous trouvez un souci, n'hésitez pas à revenir vers moi.

**Reply by JLBWai** [reply]
Bonjour Amauri, 

Tout semble bien fonctionner avec la version Piano. 
Par contre, une fois l'opt-in donné, mes tags (gérés par GTM) s'envoient en double : 
- via la config GTM 
- via TAC. 
J'aimerais que les tags ne s'envoient pas via TAC. 
Est-ce que c'est possible de bloquer l'envoi des tags dans le script ?

Merci

**Reply by AmauriC** [reply]
Il faut passer l'option `tarteaucitron.user.pianoSendData = false;` à `false` pour que tarteaucitron n'envoi pas les tags.

**Reply by JLBWai** [reply]
Cette option est déjà paramétrée, mais cela n'empêche pas l'envoi des tags : 
![TAC](https://github.com/AmauriC/tarteaucitron.js/assets/135860212/b63f0a52-b12d-4109-a155-2874ccef97d6)

**Reply by AmauriC** [reply]
Possible d'avoir l'url d'une page qui envoi en double?

**Reply by JLBWai** [reply]
Exemple sur ce site : https://preprod.maximo.fr/
Pour le marquage Piano, j'utilise l'extension Tag Inspector

**Reply by AmauriC** [reply]
Je n'ai pas accès au site :(

**Reply by JLBWai** [reply]
Est-ce que vous pouvez me partager votre adresse IP ? Je vais demander un accès

**Reply by AmauriC** [reply]
Oui bien sûr mais pas ici, contactez moi sur amauri@champeaux.fr ;)

**Reply by JLBWai** [reply]
Merci. Je vous ai envoyé un mail !

**Comment by mboumeddine** [comment]
Bonjour Amaury, 
j'ai essayé d'ajouter Piano mais j'ai rien qui s'affiche sur tarteaucitron, pas de soucis avec Google Analytics / Matomo .. etc
Code utilisé : 
`tarteaucitron.user.pianoSite = 123456;
        tarteaucitron.user.pianoCollectDomain = 'domaine.fr';
        tarteaucitron.user.pianoSendData = true;
        (tarteaucitron.job = tarteaucitron.job || []).push('pianoanalytics');`
        
Est ce que j'ai oublié quelque chose ?

Merci d'avance

**Reply by AmauriC** [reply]
Au niveau de la configuration, tout est bon. Quel est l'url du site? Le problème viens d'ailleurs ; )

**Reply by mboumeddine** [reply]
Voila un exemple d'url : 
https://www.francealumni.fr/fr?piano=1

**Reply by AmauriC** [reply]
C'est une version de 2020, ce service n'existe pas dans cette version.

Mettez à jour les fichiers tarteaucitron et tout sera ok ;)

**Reply by mboumeddine** [reply]
Super, merci beaucoup
